### DISCUSSION TITLE
Colorama and Markdown dependents

### DISCUSSION THREAD

**Discussion by James-151199** [SystemRDL/systemrdl-compiler#150] [Q&A] [answered]
Is there a way to install the systemRDL-compiler on Linux without the colorama and markdown dependent packages?

Are they both absolutely necessary to run the systemRDL compiler?

I'm currently running python 3.6.9 on Linux and don't see the need for these packages.
I'm working on a very strict network that only allows minimal user level installs. Installing other dependent packages is making it much more difficult.

Thank you.

**Comment by James-151199** [comment, marked as answer]
Just removed all dependencies for Colorama in messages.py and segment_map.py and it works flawlessly.

**Reply by amykyta3** [reply]
Glad you got it working!

Markdown is a soft dependency, as it is lazy-loaded only when invoked at runtime. It is not directly required by the compiler, but is heavily used by anything that will generate HTML documentation such as [PeakRDL-html](https://github.com/SystemRDL/PeakRDL-html)

Colorama is used to provide colorful console messages. Not strictly required, but makes things much easier to look at.
